Mathlib/RingTheory/RingHom/StandardSmooth.lean

```diff
@@ -0,0 +1,168 @@
+/-
+Copyright (c) 2024 Christian Merten. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Christian Merten
+-/
+import Mathlib.RingTheory.LocalProperties.Basic
+import Mathlib.RingTheory.Smooth.StandardSmooth
+import Mathlib.Tactic.Algebraize
+
+/-!
+# Standard smooth ring homomorphisms
+
+In this file we define standard smooth ring homomorphisms and show their
+meta properties.
+
+## Notes
+
+This contribution was created as part of the AIM workshop "Formalizing algebraic geometry"
+in June 2024.
+
+-/
+
+universe t t' w w' u v
+
+variable (n m : ℕ)
+
+open TensorProduct
+
+namespace RingHom
+
+variable {R : Type u} {S : Type v} [CommRing R] [CommRing S]
+
+/-- A ring homomorphism `R →+* S` is standard smooth if `S` is standard smooth as `R`-algebra. -/
+def IsStandardSmooth (f : R →+* S) : Prop :=
```

Comment:
```suggestion
@[algebraize]
def IsStandardSmooth (f : R →+* S) : Prop :=
```

Reply:
Did you see my above message? This causes in all uses of `algebraize` below:
```
AppBuilder for 'mkAppOptM', result contains metavariables
  Algebra.IsStandardSmooth S T
```
(I think this is due to `IsStandardSmooth` depending on additional universe levels)